app/code/Magento/Sales/ViewModel/OrderHistoryComment.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\Sales\ViewModel;
+
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+use Magento\Sales\Model\Order\Status\History;
+use Magento\Sales\Model\ResourceModel\Order\Status\History\CollectionFactory;
+
+/**
+ * View model to add order history comment to Order View page
+ */
+class OrderHistoryComment implements ArgumentInterface
+{
+    /**
+     * @var CollectionFactory
+     */
+    private $collectionFactory;
+
+    /**
+     * @param CollectionFactory $collectionFactory
+     */
+    public function __construct(
+        CollectionFactory $collectionFactory
+    ) {
+        $this->collectionFactory = $collectionFactory;
+    }
+
+    /**
+     * Return collection of not visible on frontend order status history items.
+     *
+     * @param int $orderId
+     *
+     * @return History[]
+     */
+    public function getOrderComments(int $orderId): array
+    {
+        $history = [];
+        $collection = $this->collectionFactory->create();
+        $collection->setOrderFilter($orderId)
+            ->setOrder('created_at', 'desc')
+            ->setOrder('entity_id', 'desc');
+
+        foreach ($collection as $status) {
+            if (!$status->isDeleted() && $status->getComment() && !$status->getIsVisibleOnFront()) {
```

Comment:
Could you explain why should we filter the only comments not visible on the frontend ?

Reply:
As to me it's absolutely logically to see all the comments both hidden and visible on the frontend

Reply:
@ihor-sviziev on frontend customer see in message that this order has completed from admin. And doesn't see second message with admin name. In admin order view we see different message with admin name who made this order. I see logic in this not to show exactly admin user name for customer and to see for other admin users.

Reply:
@sdzhepa @SilinMykola, it doesn't make any sense to me. Let's discuss this on stage.